toolchains/spack.gyselalibxx_env/gyselalibxx-env-0.23.0.yaml

```
@@ -0,0 +1,45 @@
+spack:
+  concretizer:
+    unify: true
+  definitions:
+  - compilers:
+    - 'gcc@11:'
+  - packages:
+    - 'benchmark'
+    - 'cmake@3.22:3'
+    - 'eigen'
+    - 'fftw precision=float,double'
+    - 'gdb ~debuginfod'
+    - 'ginkgo@1.8'
+    - 'googletest +gmock'
+    - 'openmpi'
+    - 'paraconf'
+    - 'pdi@1.6:1'
+    - 'pdiplugin-decl-hdf5 +mpi'
+    - 'pdiplugin-decl-netcdf +mpi'
+    - 'pdiplugin-mpi'
+    - 'pdiplugin-set-value'
+    - 'pdiplugin-trace'
+  - python-packages:
+    - 'py-dask'
+    - 'py-h5netcdf'
+    - 'py-h5py'
+    - 'py-imageio'
+    - 'py-matplotlib'
+    - 'py-netcdf4 +mpi'
+    - 'py-numpy'
+    - 'py-scipy'
+    - 'py-sympy'
+    - 'py-xarray'
+  specs:
+  - lapack
+  - matrix:
+    - [$packages]
+    - [$%compilers]
+  - matrix:
+    - [$python-packages]
+    - [$%compilers]
+  view: true
+  packages:
+    all:
+      variants: ~mpi ~openmp cxxstd=17
```

```suggestion
spack:
  concretizer:
    unify: true
  definitions:
  - compilers:
    - 'gcc@11'
  - packages:
    - 'benchmark@1.8'
    - 'cmake@3.22:3'
    - 'eigen@3.4'
    - 'fftw@3.3 precision=float,double'
    - 'ginkgo@1.8'
    - 'googletest@1.14 +gmock'
    - 'openmpi@5.0'
    - 'paraconf@1.0'
    - 'pdi@1.8'
    - 'pdiplugin-decl-hdf5@1.8 +mpi'
    - 'pdiplugin-decl-netcdf@1.8 +mpi'
    - 'pdiplugin-mpi@1.8'
    - 'pdiplugin-set-value@1.8'
    - 'pdiplugin-trace@1.8'
  - python-packages:
    - 'py-dask@2024.7'
    - 'py-h5netcdf'
    - 'py-h5py@3.12'
    - 'py-imageio@2.35'
    - 'py-matplotlib@3.9'
    - 'py-netcdf4@1.7 +mpi'
    - 'py-numpy@2.1'
    - 'py-scipy@1.14'
    - 'py-sympy@1.13'
    - 'py-xarray@2024.7'
  specs:
  - lapack
  - matrix:
    - [$packages]
    - [$%compilers]
  - matrix:
    - [$python-packages]
    - [$%compilers]
  view: true
  packages:
    all:
      variants: ~mpi ~openmp cxxstd=17
```

Could be someting like this, it comes from Persee config. I have removed `gdb`, I am not sure it is a good idea to try to install it, this caused an issue on macOS last time. I did not find `py-h5netcdf` on Persee.

---

We could actually make sure this setup compiles fines in the CI ? both on macOS and Ubuntu 22/24 ?

---

I was planning to add a test to the CI. I hadn't thought about mac but that is a very good idea

---

> I was planning to add a test to the CI. I hadn't thought about mac but that is a very good idea

I don't think we will be able to already run on macOS. I think we need PDI 1.9 and maybe Ginkgo 1.9.

---

1. Package py-xarray cannot satisfy '%gcc@11'
     2. Package py-sympy cannot satisfy '%gcc@11'
     3. Package py-scipy cannot satisfy '%gcc@11'
     4. Package py-numpy cannot satisfy '%gcc@11'
     5. Package py-netcdf4 cannot satisfy '%gcc@11'
     6. Package py-matplotlib cannot satisfy '%gcc@11'
     7. Package py-imageio cannot satisfy '%gcc@11'
     8. Package py-h5py cannot satisfy '%gcc@11'
     9. Package py-h5netcdf cannot satisfy '%gcc@11'
    10. Package py-dask cannot satisfy '%gcc@11'
    11. Package pdiplugin-trace cannot satisfy '%gcc@11'
    12. Package pdiplugin-set-value cannot satisfy '%gcc@11'
    13. Package pdiplugin-mpi cannot satisfy '%gcc@11'
    14. Package pdiplugin-decl-netcdf cannot satisfy '%gcc@11'
    15. Package pdiplugin-decl-hdf5 cannot satisfy '%gcc@11'
    16. Package pdi cannot satisfy '%gcc@11'
    17. Package paraconf cannot satisfy '%gcc@11'
    18. Package openmpi cannot satisfy '%gcc@11'
    19. Package googletest cannot satisfy '%gcc@11'
    20. Package ginkgo cannot satisfy '%gcc@11'
    21. Package fftw cannot satisfy '%gcc@11'
    22. Package eigen cannot satisfy '%gcc@11'
    23. Package cmake cannot satisfy '%gcc@11'
    24. Package benchmark cannot satisfy '%gcc@11'. You could consider setting `concretizer:unify` to `when_possible` or `false` to allow multiple versions of some packages.

---

Hmm, it does say:
> ==> Fetching https://ghcr.io/v2/spack/bootstrap-buildcache-v1/blobs/sha256:af305366a56ff47603977201d86a296cfaf6f5e6ed38412ecd68efb3e8ad9799

But I'm not sure why

---

Bootstrapping phase, I guess it is ok